cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -435,11 +436,12 @@ private void httpExceptionHandler(Promise<? extends CruiseControlResponse> resul
             // Vert.x throws a NoStackTraceTimeoutException (inherits from TimeoutException) when the request times out
             // so we catch and raise a TimeoutException instead
             result.fail(new TimeoutException(t.getMessage()));
-        } else if (t instanceof ConnectException) {
+        } else if (t instanceof NoRouteToHostException || t instanceof ConnectException) {
+            // Netty throws a AnnotatedNoRouteToHostException (inherits from NoRouteToHostException) and  when it cannot resolve the host
             // Vert.x throws a AnnotatedConnectException (inherits from ConnectException) when the request times out
-            // so we catch and raise a ConnectException instead
-            result.fail(new ConnectException(t.getMessage()));
-        } else {
+            // so we catch and raise a CruiseControlRetriableConnectException instead since
+            result.fail(new CruiseControlRetriableConnectException(t.getMessage()));
```

Comment:
So I think this is the right approach. However, now the KRAO is expecting **any** implementation of the `CruiseControlApi` interface to fail a future, if it is a transient connection error, with a `CruiseControlRetriableConnectException` from many/all its methods. 

Ideally, we would put this exception handler in an an abstract class so other implementation could use it, but maybe it would be acceptable to simply update the documentation of the interface class to explicitly say that any transient connection error should be failed with a `CruiseControlRetriableConnectException`?

@ppatierno WDYT?

Reply:
>  other implementation could use it

Which "other implementation" are you referring to?